--- a/src/desktop/main.c
+++ b/src/desktop/main.c
@@ -1523,6 +1523,7 @@
             globalInputRecording->filterDebugKeys = args.debug;
             installCrashHandlers();
         }
+#ifdef ENABLE_VM_TRACING
         shcopyFromTo(args.varReadsToBeTraced, runner->vmContext->varReadsToBeTraced);
         shcopyFromTo(args.varWritesToBeTraced, runner->vmContext->varWritesToBeTraced);
         shcopyFromTo(args.functionCallsToBeTraced, runner->vmContext->functionCallsToBeTraced);
@@ -1534,6 +1535,7 @@
         shcopyFromTo(args.stackToBeTraced, runner->vmContext->stackToBeTraced);
         shcopyFromTo(args.tilesToBeTraced, runner->vmContext->tilesToBeTraced);
         runner->vmContext->traceBytecodeAfterFrame = args.traceBytecodeAfterFrame;
+#endif
         runner->vmContext->alwaysLogUnknownFunctions = args.alwaysLogUnknownFunctions;
         runner->vmContext->alwaysLogStubbedFunctions = args.alwaysLogStubbedFunctions;
         runner->vmContext->traceEventInherited = args.traceEventInherited;
